src/SFA.DAS.EmployerIncentives.Functions.LegalEntities/Services/LegalEntities/IEmployerVendorIdService.cs

```diff
@@ -0,0 +1,9 @@
+﻿using System.Threading.Tasks;
+
+namespace SFA.DAS.EmployerIncentives.Functions.LegalEntities.Services.LegalEntities
+{
+    public interface IEmployerVendorIdService
+    {
+        Task AddEmployerVendorId(string hashedLegalEntityId);
```

Comment:
Should this be just called Add?  Since it applied to the EmployerVendorIdService.  Seems to be a tautology.  In fact it is confusing because you are not adding a vendorId but a hashedLegalentityId. Which raises the question is this the IEmployerVendorService?  I think it would be better to add an object that contains the hashedLegalentityId then it's open to adding other data without changing the interface or having multiple Methods for each use case.

Reply:
It's triggering a call to the Outer API to add the VRF vendor ID for the given legal entity ID, I think.

Reply:
It's a very confusing interface.

Reply:
Perhaps we should change the name to GetAndAddEmployerVendorId() as it's calling BC to get the EmployerVendorId and adding that value to the LegalEntity

Reply:
Except it's not getting anything (i.e. not returning anything)

Reply:
Perhaps we should therefore call it AddEmployerVendorIdLegalEntity("XXX"). This is more explicit

Reply:
I did suggest one.  Add(...) except you'd have to add an EmployerVendorId employerVendorId)

where EmployerVendorId is

class EmployerVendorId
{
  string hashedLegalEntityId
}

Since then it is a service for EmployerVendorId's which is what it's name implies.